generator/lib/src/realm_field_info.dart

```diff
@@ -60,11 +61,13 @@ class RealmFieldInfo {
   Iterable<String> toCode() sync* {
     yield '@override';
     yield "$typeName get $name => RealmObject.get<$basicTypeName>(this, '$realmName') as $typeName;";
-    if (!isFinal) {
+    bool generateSetter = !isFinal && !primaryKey && !isRealmCollection;
+    if (generateSetter) {
       yield '@override';
       yield "set $name(${typeName != typeModelName ? 'covariant ' : ''}$typeName value) => RealmObject.set(this, '$realmName', value);";
     } else {
-      if (isLate) { // we still need to override setter on late final, but just throw error!
+      bool generateThrowError = isLate || primaryKey || isRealmCollection;
+      if (generateThrowError) { // we still need to override setter on late final, but just throw error!
```

Comment:
Have you folks looked into how migrations are going to be done? Because during a migration, the user should be able to change the primary key. This is how the PK setter looks like in C# to handle this use case: https://github.com/realm/realm-dotnet/blob/aba49e08af41066389f34cc18169976e2e1b7e05/Realm/Realm/DatabaseTypes/RealmObjectBase.cs#L223-L235

Reply:
no we have not reach that point. We will change the generator accordingly for that.
thanks for the heads up.